File: data-augmentation/server/models/Repository.js
const mongoose = require('mongoose');

const RepositorySchema = new mongoose.Schema({
    projectName: String,
    githubLink: String,
    commit: String,
    srcPathList: [String],
    rootPathList: [String],
    classes: [{
        _id: String,
        name: String
    }]
})

exports.Repository = mongoose.model('Repository', RepositorySchema);

File: data-augmentation/src/main/java/star/tratto/records/Repository.java
package star.tratto.records;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record Repository(
        @JsonProperty("projectName") String projectName,
        @JsonProperty("githubLink") String githubLink,
        @JsonProperty("commit") String commit,
        @JsonProperty("srcPathList") List<String> srcPathList,
        @JsonProperty("rootPathList") List<String> rootPathList
) {}
